### PR TITLE
Add page for JobSpec, unit and ui tests, error.html for handling errors

### PR DESCRIPTION
**Pull request by RemyKainos**

### CODE REVIEW

File: src/service/JobSpecService.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+const jobSpecService = {
```
Comment (TomBillingtonUK) [inline src/service/JobSpecService.ts:3]: This should be a class with methods and not a const

File: src/views/ViewJobSpec.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{% extends "base.html" %}
```
Comment (TomBillingtonUK) [inline src/views/ViewJobSpec.html:1]: Can you add a screenshot of this page working to the PR

File: src/views/error.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{% extends "base.html" %}
```
Comment (TomBillingtonUK) [inline src/views/error.html:1]: Can you add a Image of this page to the PR

File: src/views/error.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+{% block content %}
+    <main class="main-content">
+        <h1 class="error-text">{{data.status}}: {{data.message}}</h1>
```
Comment (TomBillingtonUK) [inline src/views/error.html:5]: The way this is setup is that the error from the API or whatever exception throw will be displayed on the screen. You want to display a user friendly message and log the actual error to the console

File: test/ui/JobSpecTest.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import webdriver from 'selenium-webdriver';
+
+import chai from 'chai';  
+
+describe('Job Spec Test', async () => {
+
+
+    it('View job spec', async () => {
```
Comment (TomBillingtonUK) [inline test/ui/JobSpecTest.ts:8]: Merge main into this PR and check if the tests still pass. Now login is enabled you may need to login to view this page going forward

**Review by CallumE74 — APPROVED** [review]

**Review by TomBillingtonUK — APPROVED** [review]

**Comment by RemyKainos** [comment]
Screenshot 2023-09-25 at 13 43 00

**Pull request merged**